server/webapp/WEB-INF/rails.new/webpack/views/dashboard/dashboard_widget.js.msx

```diff
@@ -92,6 +93,99 @@ const DashboardWidget = {
                                             vm={vnode.state.pipelineSelectionVM}/>);
     }
 
+    let groupByEnvironment;
+
+    if (vm.groupByEnvironment.visible()) {
+      groupByEnvironment = <f.checkbox model={vm.groupByEnvironment} attrName="enabled" listener="onclick" label="Group by Environment?" />;
+    }
+
+    function renderEnvironments(pipelinesByEnvironment) {
+      const envs = Object.keys(pipelinesByEnvironment).sort();
+
+      if (envs.indexOf("*") > -1) {
+        envs.splice(envs.indexOf("*"), 1);
+        envs.push("*");
+      }
```

Comment:
The "*" stuff seems a bit hacky to me. Maybe it would be more readable if we had a separate list to maintain pipelines without an environment. Thoughts?

Reply:
It could also arguably be `null` instead of a magic string, which would be more accurate. I probably didn't want to null check while sorting (don't remember why I chose this). I sort of prefer keeping everything in the same collection rather than a separate list because then we can use the same code. I would rather not have to write a separate case to handle the list of ungrouped pipelines.

Changing the ungrouped pipelines to live under a `null` key in `pipelinesByEnvironment` could simplify this to make it more readable as well. I just tested in the console; sorting an array of environment names always puts `null` at the bottom, which is what I was going for anyway. Also, the below lines (e.g. setting `name` and `settingsIcon` can be simplified with something like `name = env || "(no environment)"`.

What do you think?